osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs

```diff
@@ -262,24 +230,16 @@ private void updateLayoutRecursive(DrawableHitObject hitObject)
 
         private void updatePosition(DrawableHitObject hitObject, double currentTime)
         {
-            switch (direction.Value)
-            {
-                case ScrollingDirection.Up:
-                    hitObject.Y = scrollingInfo.Algorithm.PositionAt(hitObject.HitObject.StartTime, currentTime, timeRange.Value, scrollLength);
-                    break;
+            float position = PositionAtTime(hitObject.HitObject.StartTime, currentTime);
 
-                case ScrollingDirection.Down:
-                    hitObject.Y = -scrollingInfo.Algorithm.PositionAt(hitObject.HitObject.StartTime, currentTime, timeRange.Value, scrollLength);
-                    break;
-
-                case ScrollingDirection.Left:
-                    hitObject.X = scrollingInfo.Algorithm.PositionAt(hitObject.HitObject.StartTime, currentTime, timeRange.Value, scrollLength);
-                    break;
+            // The position returned from `PositionAtTime` is assuming the `TopLeft` anchor.
+            // A correction is needed because the hit objects are using a different anchor for each direction (e.g. `BottomCentre` for `Bottom` direction).
+            float anchorCorrection = axisInverted ? scrollLength : 0;
```

Comment:
This cannot actually `hitObject.AnchorPosition` due to nested hit objects.

Reply:
Is that something that's blocking this pull from being merged, or some kind of a general note? Hard to tell what the intention of this comment is supposed to be.

Reply:
It is just a note.

Reply:
I was also similarly confused by this xmldoc, so I've reworded it a touch and added an example to hopefully explain better. See if you think that's more clear than what was here before.

(Disabling auto-merge pending confirmation.)